src/scaffoldmaker/meshtypes/meshtype_3d_brainstem.py

```diff
@@ -0,0 +1,355 @@
+
+        return annotationGroups
+
+    @classmethod
+    def refineMesh(cls, meshRefinement, options):
+        """
+        Refine source mesh into separate region, with change of basis.
+        :param meshRefinement: MeshRefinement, which knows source and target region.
+        :param options: Dict containing options. See getDefaultOptions().
+        """
+        assert isinstance(meshRefinement, MeshRefinement)
+        refineElementsCountAcrossMajor = options['Refine number of elements across major and minor']
+        refineElementsCountAlong = options['Refine number of elements along']
+        meshRefinement.refineAllElementsCubeStandard3d(refineElementsCountAcrossMajor, refineElementsCountAlong, refineElementsCountAcrossMajor)
+
+    @classmethod
+    def defineFaceAnnotations(cls, region, options, annotationGroups):
+        """
+        Add face annotation groups from the 1D mesh.
+        :param region: Zinc region containing model.
+        :param options: Dict containing options. See getDefaultOptions().
+        :param annotationGroups: List of annotation groups for ventral-level elements.
+        New point annotation groups are appended to this list.
+        """
+        # create  groups
+        fm = region.getFieldmodule()
+        mesh2d = fm.findMeshByDimension(2)
+        is_exterior = fm.createFieldIsExterior()
+        is_exterior_face_xi1 = fm.createFieldOr(
+            fm.createFieldAnd(is_exterior, fm.createFieldIsOnFace(Element.FACE_TYPE_XI1_0)),
+            fm.createFieldAnd(is_exterior, fm.createFieldIsOnFace(Element.FACE_TYPE_XI1_1)))
+        is_exterior_face_xi3 = fm.createFieldOr(fm.createFieldAnd(is_exterior, fm.createFieldIsOnFace(Element.FACE_TYPE_XI3_0)), fm.createFieldAnd(is_exterior, fm.createFieldIsOnFace(Element.FACE_TYPE_XI3_1)))
+
+        annoGroup = AnnotationGroup(region, get_brainstem_annotation_term('brainstem'))
+        isGroup = annoGroup.getFieldElementGroup(mesh2d)
+        is_face1 = fm.createFieldAnd(isGroup, is_exterior_face_xi1)
+        is_face3 = fm.createFieldAnd(isGroup, is_exterior_face_xi3)
+        is_face_ext = fm.createFieldOr(is_face1, is_face3)
+        faceGroup = findOrCreateAnnotationGroupForTerm(annotationGroups, region, ("brainstem_exterior", None))
+        faceGroup.getMeshGroup(mesh2d).addElementsConditional(is_face_ext)
+
+        # external regions
+        namelist = ['midbrain', 'pons', 'medulla oblongata']
+        for subregion in namelist:
+            subGroup = AnnotationGroup(region, get_brainstem_annotation_term(subregion))
+            issub = subGroup.getFieldElementGroup(mesh2d)
+            is_subface = fm.createFieldOr(fm.createFieldAnd(issub, is_exterior_face_xi1), fm.createFieldAnd(issub, is_exterior_face_xi3))
+            subFaceGroup = findOrCreateAnnotationGroupForTerm(annotationGroups, region, (subregion+'_exterior', None))
+            subFaceGroup.getMeshGroup(mesh2d).addElementsConditional(is_subface)
```

Comment:
Always end files with new line character - can see the warning symbol.
It's a coding standard thing.